crates/ruff/src/rules/flake8_comprehensions/rules/unnecessary_map.rs

```diff
@@ -103,11 +103,19 @@ pub(crate) fn unnecessary_map(
                 return;
             };
 
-            if parameters
-                .as_ref()
-                .is_some_and(|parameters| late_binding(parameters, body))
-            {
-                return;
+            if let Some(parameters) = parameters.as_ref() {
+                if late_binding(parameters, body) {
+                    return;
+                }
+
+                parameters
+                    .posonlyargs
+                    .iter()
+                    .chain(&parameters.args)
+                    .chain(&parameters.kwonlyargs)
+                    .any(|param| param.default.is_some())
```

Comment:
Would it be useful to have this as a method on `Parameters` node? (`has_defaults`?)

Reply:
I think what we have here is reasonable, maybe if we find ourselves doing this in at least one other place.